src/profile-logic/process-profile.js

```diff
@@ -835,14 +835,35 @@ export function processProfile(
  * Take a processed profile and remove any non-serializable classes such as the
  * StringTable class.
  */
-export function serializeProfile(profile: Profile): string {
+export function serializeProfile(
+  profile: Profile,
+  includeNetworkUrls: boolean = true
+): string {
   // stringTable -> stringArray
   const newProfile = Object.assign({}, profile, {
     threads: profile.threads.map(thread => {
-      const stringTable = thread.stringTable;
+      const stringArray = thread.stringTable.serializeToArray();
       const newThread = Object.assign({}, thread);
       delete newThread.stringTable;
-      newThread.stringArray = stringTable.serializeToArray();
+      if (includeNetworkUrls === false) {
+        for (let i = 0; i < newThread.markers.data.length; i++) {
+          const currentMarker = newThread.markers.data[i];
+          if (
+            currentMarker &&
+            currentMarker.type &&
+            currentMarker.type === 'Network'
+          ) {
+            // Remove the URI fields from marker payload.
+            currentMarker.URI = '';
+            currentMarker.RedirectURI = '';
+            // Strip the URL from the marker name
+            stringArray[newThread.markers.name[i]] = stringArray[
+              newThread.markers.name[i]
+            ].split(':')[0];
```

Comment:
Can you slice using `indexOf(':')` or a regexp instead ? The goal is to avoid instancianting an array. Also please use a temporary variable for the string index: For example:
```javascript
const stringIndex = newThread.markers.name[i];
stringArray[stringIndex] = stringArray[stringIndex].replace(/:.*/, '');
```
Or:
```javascript
const stringIndex = newThread.markers.name[i];
stringArray[stringIndex] = stringArray[stringIndex].slice(0, stringArray[stringIndex].indexOf(':'));
```
(I prefer the conciseness of the first proposal :) )

Reply:
Yeah, the first one looks better. Thanks for the suggestion.